views/json/resources/default.php

```diff
@@ -0,0 +1,3 @@
+<?php
+
+echo json_encode($vars);
```

Comment:
This would also allow us to use absolute URLs instead of AJAX views that keep failing with non-XHR requests. We could hook into the AJAX API, and only use layout elements that we care about.